site/content/blog/carvelizing-helm-chart.md

```diff
@@ -0,0 +1,293 @@
+  name: simple-package-repository
+spec:
+  fetch:
+    imgpkgBundle:
+      image: ${REPO_HOST}/packages/nginx-bitnami-repo:1.0.0
+```
+
+```bash
+$ kapp deploy -a repo -f repo.yaml -y
+```
+
+After deploying, wait for the packageRepository description to become `Reconcile succeeded`. 
+
+```bash 
+$ kubectl get packagerepository
+```
+
+Now, we can see the list of package metadata's available
+
+```bash
+$ kubectl get packagemetadatas
+```
+
+**2. List Packages**:
+We can see the list of packages and their version's available for install 
+
+```bash 
+kubectl get packages
+```
+
+As we can see, our published nginx helm package is available for us to install.
+
+**3. Create Service Account**: To install the above package, we need to create default-ns-sa service account that give PackageInstall CR privileges to create resources in the default namespace
+
+```bash
+$ kapp deploy -a default-ns-rbac -f https://raw.githubusercontent.com/vmware-tanzu/carvel-kapp-controller/develop/examples/rbac/default-ns.yml -y
+```
+
+**4. Install the Package**: To install a carvel Package, we need to create PackageInstall Kuberentes resource. A Package Install will install the nginx helm package and its underlying resources on a Kubernetes cluster. A `PackageInstall` references a `Package`. Thus, we can create the `PackageInstall` yaml from the `Package`.
+
+We are providing our custom values via secret. 
+
+**NOTE**: If you are using minikube, for nginx service to be in `ACTIVE` state, start `minikube tunnel` in other window as services of LoadBalancer types do not come up otherwise in minikube.  
+
+Save the below PackageInstall CR to a file named `pkginstall.yaml`
+```yaml
+---
+apiVersion: packaging.carvel.dev/v1alpha1
+kind: PackageInstall
+metadata:
+  name: nginx-pkg
+spec:
+  serviceAccountName: default-ns-sa
+  packageRef:
+    refName: nginx.bitnami.vmware.com
+    versionSelection:
+      constraints: 1.0.0
+  values:
+  - secretRef:
+      name: nginx-values
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: nginx-values
+stringData:
+  values.yml: |
+    ---
+    image:
+      pullPolicy: Always
+    serverBlock: |-
+      server {
+        listen 0.0.0.0:8080;
+        location / {
+          return 200 "Response from Custom Server";
+        }
+      }
+```
+
+```bash
+$ kapp deploy -a pkg-demo -f pkginstall.yaml -y
+```
+
+After the deploy has finished, kapp-controller will have installed the package in the cluster. We can verify this by checking the pods to see that we have a workload pod running. The output should show a single running pod which is part of nginx.
+
+```bash
+$ kubectl get pods
+```
+
+Once the pod is ready, you can use kubectl’s port forwarding to verify the customized response used in the workload.
+
+```bash
+$ kubectl port-forward service/nginx-pkg 3000:80 &
+```
+
+Now if we make a request against our service, we can see that server response is ```Response from Custom Server```
+
+```bash 
+$ curl localhost:3000
+```
+
```

Comment:
maybe a quick Conclusion sentence/section before the "join" call to action?

## Conclusion
You have now seen how to wrap an existing Helm chart in a Carvel package. 

or similar.

Reply:
Adopted